### PR TITLE
Accessibility Fixes for Several SQL Based Datasource Config Pages

### LINKED ISSUE #42717: Accessibility: Fix MSSQL config page fastpass issues

### PR DESCRIPTION
**Pull request by codeincarnate**
#### Overview

Contains updates for SQL based data source config pages (MySql, Postgres, MSSQL) uncovered with [Fastpass](https://accessibilityinsights.io/docs/en/web/getstarted/fastpass/). Main error was a missing label on password select fields.

#### Fixes

#42717, #42719, #42723

#### Reviewer Notes

Additional attention to changes in `ng_react.ts` would be appreciated. [ARIA attributes in React](https://reactjs.org/docs/accessibility.html) are not camel cased. The fix applied here would allow for non camel cased attributes to be applied when using React components as Angular directives.

### CODE REVIEW

**Review by mdvictor — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/mssql/partials/config.html [hunk 1]
```diff
@@ -13,9 +13,9 @@ <h3 class="page-heading">MS SQL connection</h3>
 	</div>
 
 	<div class="gf-form">
-		<label class="gf-form-label width-7">Authentication</label>
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mssql/partials/config.html:16]: I would probably use this the other way around. Adding the id to the select and using a for attribute here.

File: public/app/plugins/datasource/mysql/partials/config.html [hunk 2]
```diff
@@ -23,6 +23,7 @@ <h3 class="page-heading">MySQL Connection</h3>
         on-reset="ctrl.onPasswordReset"
         on-change="ctrl.onPasswordChange"
         inputWidth="9"
+		aria-label="'Password'"
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mysql/partials/config.html:26]: this is really just nitpicking but can you please align it with the other attributes?

File: public/app/plugins/datasource/postgres/partials/config.html [hunk 3]
```diff
@@ -25,16 +25,17 @@ <h3 class="page-heading">PostgreSQL Connection</h3>
...
         on-change="ctrl.onPasswordChange"
         inputWidth="9"
+        aria-label="'Password'"
       />
     </div>
   </div>
 
   <div class="gf-form">
-    <label class="gf-form-label width-10">TLS/SSL Mode</label>
+    <label class="gf-form-label width-10" id="tls-mode-label">TLS/SSL Mode</label>
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/postgres/partials/config.html:34]: Can you please change it here as well?

File: public/app/plugins/datasource/postgres/partials/config.html [hunk 4]
```diff
@@ -25,16 +25,17 @@ <h3 class="page-heading">PostgreSQL Connection</h3>
         on-reset="ctrl.onPasswordReset"
         on-change="ctrl.onPasswordChange"
         inputWidth="9"
+        aria-label="'Password'"
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/postgres/partials/config.html:28]: I wonder if we need double quotes there?
```suggestion
        aria-label="Password"
```

File: public/app/plugins/datasource/mysql/partials/config.html [hunk 5]
```diff
@@ -23,6 +23,7 @@ <h3 class="page-heading">MySQL Connection</h3>
         on-reset="ctrl.onPasswordReset"
         on-change="ctrl.onPasswordChange"
         inputWidth="9"
+        aria-label="'Password'"
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mysql/partials/config.html:26]:
```suggestion
        aria-label="Password"
```

File: public/app/plugins/datasource/mssql/partials/config.html [hunk 6]
```diff
@@ -37,14 +37,15 @@ <h3 class="page-heading">MS SQL connection</h3>
...
 				labelWidth="7"
 				inputWidth="7"
+				aria-label="'Password'"
 			/>
 		</div>
 	</div>
 
 	<div class="gf-form">
-		<label class="gf-form-label width-7">Encrypt</label>
+		<label class="gf-form-label width-7" id="encrypt-label">Encrypt</label>
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mssql/partials/config.html:46]: Can you please change it here as well?

File: public/app/plugins/datasource/mssql/partials/config.html [hunk 7]
```diff
@@ -37,14 +37,15 @@ <h3 class="page-heading">MS SQL connection</h3>
 				on-change="ctrl.onPasswordChange"
 				labelWidth="7"
 				inputWidth="7"
+				aria-label="'Password'"
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mssql/partials/config.html:40]:
```suggestion
				aria-label="Password"
```

File: public/app/angular/services/ng_react.ts [hunk 8]
```diff
@@ -168,9 +168,9 @@ function getPropExpression(prop: any) {
  * @param attrs All attributes of the component.
  * @param propName Name of the prop that react component expects.
  */
-function findAttribute(attrs: string, propName: string): string {
-  const index = Object.keys(attrs).find((attr) => {
-    return attr.toLowerCase() === propName.toLowerCase() || attr.toLowerCase() === kebabCase(propName);
+function findAttribute(attrs: object, propName: string): string {
+  const index = Object.keys(attrs).find((attr: any) => {
+    return attr.toLowerCase() === propName.toLowerCase() || kebabCase(attr) === kebabCase(propName);
```
Comment (dprokop) [inline public/app/angular/services/ng_react.ts:173]: Would be nice to have a test for this.

**Review by dprokop — APPROVED** [review]
LGTM

File: public/app/angular/services/ng_react.ts (hunk 8 above)
Comment (codeincarnate) [reply public/app/angular/services/ng_react.ts:173]: I'll create a follow up PR with the test for this one 😄

**Pull request merged**